### PR TITLE
Unite web and mobile welcome page

### PR DESCRIPTION
**Pull request by BeatC**
In this PR I move logic for `WelcomePage` component from jQuery app to React one. Added room name generation to `AbstractWelcomePage` for further using it in RN version of app.

### CODE REVIEW

File: modules/UI/welcome_page/WelcomePage.js [hunk 1]
```diff
@@ -1,68 +1,20 @@
-/* global $, interfaceConfig, APP */
-var animateTimeout, updateTimeout;
-
-var RoomnameGenerator = require("../../util/RoomnameGenerator");
-import UIUtil from "../util/UIUtil";
+/* global $ */
 
 function enter_room() {
```
Comment (lyubomir) [inline modules/UI/welcome_page/WelcomePage.js]: Naming such as `enter_room` is not consistent with our conventions. Unless there's a strong reason against it, please rename it in accord with our naming conventions.

File: react/features/base/util/randomUtil.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ */
+const HEX_DIGITS = '0123456789abcdef';
+
+/**
+ * Utility singleton responsible for generation of random sequences
+ *
+ * @singleton
+ * @const
+ */
+const RandomUtil = {
```
Comment (lyubomir) [inline react/features/base/util/randomUtil.js]: RandomUtil is a collection of functions without state. Consequently, I see no strong reason for RandomUtil to exist in ES6 source code and be exported as such. Please remove RandomUtil and export each of its functions separately.

File: react/features/base/util/roomnameGenerator.js [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    '_ADJECTIVE__PLURALNOUN__VERB__ADVERB_'
+];
+
+/**
+ * Singleton object responsible for generating of new room names.
+ *
+ * @singleton
+ * @const
+ **/
+export const RoomnameGenerator = {
```
Comment (lyubomir) [inline react/features/base/util/roomnameGenerator.js]: RoomnameGenerator is a collection of functions without state. Consequently, I see no strong reason for RoomnameGenerator to exist in ES6 source code and be exported as such. Please remove RoomnameGenerator and export each of its functions separately.

File: react/features/welcome/components/AbstractWelcomePage.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { appNavigate } from '../../app';
 import { isRoomValid } from '../../base/conference';
 import { VideoTrack } from '../../base/media';
 import { getLocalVideoTrack } from '../../base/tracks';
+import { RoomnameGenerator as RoomnameGen } from '../../base/util';
```
Comment (lyubomir) [inline react/features/welcome/components/AbstractWelcomePage.js]: Using multiple names for one and the same entity is against our coding style. Please don't use a second name for the same entity and comply with our coding style.

**Comment by BeatC** [comment]
jenkins test this please

**Comment by lyubomir** [comment]
Superseded by https://github.com/jitsi/jitsi-meet/pull/1200.

**Pull request closed without merging**